frontend/src/components/App/App.js

```diff
@@ -1,18 +1,38 @@
-import React from "react";
+import {useEffect, React} from "react";
 import {
     BrowserRouter as Router,
     Switch,
     Route,
     Redirect
 } from "react-router-dom";
-import { URLS, EXPERIMENT_SLUG } from "../../config";
+import { create } from "zustand";
+import { EXPERIMENT_SLUG, URLS } from "../../config";
+import { getParticipant } from "API";
 import Experiment from "../Experiment/Experiment";
 import Profile from "../Profile/Profile";
 import Reload from "../Reload/Reload";
 import StoreProfile from "../StoreProfile/StoreProfile.js";
 
+export const useParticipantStore = create((set) => ({
+    participant: null,
+    setParticipant: (participant) => set((state) => ({participant}))
+}));
+
 // App is the root component of our application
 const App = () => {
+    const setParticipant = useParticipantStore((state) => state.setParticipant);
+    const queryParams = window.location.search;
+    
+    useEffect(() => {
+        if (queryParams && !(new URLSearchParams(queryParams).has("participant_id"))) {
+            console.error("Unknown URL parameter, use ?participant_id=")
+            return;
+        }
+        getParticipant(queryParams).then(data => {
+            setParticipant(data);
+        })
+    }, [])
```

Comment:
If you handle the error here, you could choose to show an error message if you want and prevent the experiment from being shown/loaded.